### PR TITLE
Proof read linked-list.md

### PR DESCRIPTION
**Pull request by Carloscode5**
### Description

The following is a brief explanation on [Linked Lists](https://www.codecademy.com/resources/docs/c) in the Language of [C](https://www.codecademy.com/resources/docs/c). There are four main types, though one will come to find there are many types of these data structures. These four are defined, presented in an example and provided with the proper syntax in the Language of C. 

### Type of Change

- Adding a new entry

### Checklist

- [✓] All writings are my own.
- [✓] My entry follows the Codecademy Docs style guide.
- [✓] My changes generate no new warnings.
- [✓] I have performed a self-review of my own writing and code.
- [✓] I have checked my entry and corrected any misspellings.
- [✓] I have made corresponding changes to the documentation if needed.
- [✓] I have confirmed my changes are not being pushed from my forked `main` branch.
- [✓] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [✓] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/c/concepts/linked-list/linked-list.md [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+Title: 'Linked List'
+Description: 'A data structure that contains a collection of nodes that together with the help of pointers, display a sequence.'
+Subjects:
+  - 'List Types'
+  - 'Computer Science'
+Tags:
+  - 'Singly linked lists'
+  - 'Doubly linked lists'  
+  - 'Circular linked lists'
+  - 'Circular doubly linked lists'
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:11]: Here apart from `Double Linked List` the other tags don't exist in the `tags.md` file. Please add them there or remove these tags from here and add one that is present in the `tags.md` file.

File: content/c/concepts/linked-list/linked-list.md [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+A **Linked List** is a data structure that contains a collection of nodes that together with the help of pointers, display a sequence. There are a number of list types, we will cover four of the most commmon with examples in the C language.
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:17]:
```suggestion
A **`Linked List`** is a data structure that contains a collection of `nodes` that together with the help of pointers, display a sequence. There are a number of list types, we will cover four of the most common with examples in the `C language`.
```

File: content/c/concepts/linked-list/linked-list.md [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+
+```pseudo
+#include 
+#include 
+
+typedef struct Node {
+    void* data;
+    struct Node *next;
+} Node;
+//push to the front of the list
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:29]:
```suggestion
```

File: content/c/concepts/linked-list/linked-list.md [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+struct Node* push(struct Node* head, void* data) {
+    struct Node* new_node = (struct Node*)malloc(sizeof(struct Node));
+    new_node->data = data;
+    new_node->next = head;
+    return new_node;
+}
+int main(){
+    Node* list = NULL;
+    //remember to set the list after push
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:38]:
```suggestion
```

File: content/c/concepts/linked-list/linked-list.md [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+    list = push(list, "!");
+    list = push(list, "World ");
+    list = push(list, "Hello ");
+    //print out the elements
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:42]:
```suggestion
```

File: content/c/concepts/linked-list/linked-list.md [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+    while(crnt != NULL){
+        printf("%s", crnt->data);
+        crnt = crnt->next;
+    }
+    return 0;
+}
+```
+
+## Example
+
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:53]: Give a small introductory sentence here which gives a rough idea of what one can expect from the given code snippet present below.

File: content/c/concepts/linked-list/linked-list.md [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+      printf("%d->",temp->data);
+      temp = temp->next;
+  }
+  printf("NULL");
+
+  return 0;
+}
+```
+
+## Singly linked lists
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:100]:
```suggestion
### Singly linked lists

```
Give a white space after heading and make the types as sub heading.

File: content/c/concepts/linked-list/linked-list.md [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+A singly linked list is a collection of data called nodes, where each node is divided into two parts to store data and address at some random addresses. The pointer next, points to the address of the next node in a list.
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:101]:
```suggestion
A `singly linked list` is a collection of data called `nodes`, where each `node` is divided into two parts to store `data` and `address` at some random addresses. The pointer next points to the address of the next `node` in a list.
```

File: content/c/concepts/linked-list/linked-list.md [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+  }
+
+  while (head != NULL) {
+    printf("%d\n", head->data);
+    head = head->next;
+  }
+}
+```
+
+## Doubly linked lists
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:137]:
```suggestion
### Doubly linked lists  

```

File: content/c/concepts/linked-list/linked-list.md [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+
+  while (head != NULL) {
+    printf("%d\n", head->data);
+    head = head->next;
+  }
+}
+```
+
+## Doubly linked lists  
+A doubly linked list consists of two pointers that store the address of the next successive node and the address of the previous node.
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:138]:
```suggestion
A `doubly linked list` consists of two pointers that store the address of the next successive `node` and the `address` of the previous node.
```

File: content/c/concepts/linked-list/linked-list.md [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+  insert_at_beginning(&head, 1);
+  insert_at_beginning(&head, 2);
+  insert_at_end(&head, 3);
+  insert_at_end(&head, 4);
+  print_list(head);
+  return 0;
+}
+```
+
+## Circular linked lists
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:190]:
```suggestion
### Circular linked lists

```

File: content/c/concepts/linked-list/linked-list.md [hunk 12]
```diff
@@ -0,0 +1,325 @@
...
+A circular linked list is a list in which the nodes are connected in a sequence, where the last node always points to the head (first) node of the linked list.
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:191]:
```suggestion
A `circular linked` list is a list in which the nodes are connected in a sequence, where the last node always points to the head (first) node of the `linked list`.
```

File: content/c/concepts/linked-list/linked-list.md [hunk 13]
```diff
@@ -0,0 +1,325 @@
...
+int main() {
+  insert(1);
+  insert(2);
+  insert(3);
+  print();
+  return 0;
+}
+```
+
+## Circular doubly linked lists
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:240]:
```suggestion
### Circular doubly linked lists

```

File: content/c/concepts/linked-list/linked-list.md [hunk 14]
```diff
@@ -0,0 +1,325 @@
...
+A circular doubly linked list is a data structure that consists of a collection of nodes, where each node contains a data element and two pointers: one that points to the next node in the list and another that points to the previous node in the list.
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:241]:
```suggestion
A `circular doubly linked list` is a data structure that consists of a collection of nodes, where each node contains a data element and two pointers: one that points to the next node in the list and another that points to the previous node in the list.
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
First, commit these changes and work on the comments that I've given. Once it is done I will do another review.

**Comment by dakshdeepHERE** [comment]
@Carloscode5 I have requested some changes from your side.

File: content/c/concepts/linked-list/linked-list.md [hunk 15]
```diff
@@ -0,0 +1,325 @@
+---
+Title: 'Linked List'
+Description: 'A data structure that contains a collection of nodes that together with the help of pointers, display a sequence.'
+Subjects:
+  - 'List Types'
```
Comment (dakshdeepHERE) [inline content/c/concepts/linked-list/linked-list.md:5]:
```suggestion
  - 'Code Foundations'
```
Only choose subjects from [here](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md)

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]

**Comment by dakshdeepHERE** [comment]
@Carloscode5 I'm checking in, can you commit the changes?

**Comment by Carloscode5** [comment]
Hi there, going to make the changes and commit then PR for review.Thank you for checking in.Best,Carlos ValenzuelaOn Nov 6, 2023, at 11:17 AM, Daksha Deep ***@***.***> wrote:﻿
@Carloscode5 I'm checking in, can you commit the changes?

**Comment by dakshdeepHERE** [comment]
@Carloscode5 🚀

**Comment by Carloscode5** [comment]
Hi Daksha,

Are you a part of the Codecademy review team?

Best,

Carlos Valenzuela

**Comment by dakshdeepHERE** [comment]
@Carloscode5 Yes, Let me look through the entry again. 🚀

**Comment by dakshdeepHERE** [comment]
@Carloscode5 can you commit the requested changes? So we can move forward with reviews.

File: content/c/concepts/linked-list/linked-list.md (hunk 1 above)
Comment (Carloscode5) [reply content/c/concepts/linked-list/linked-list.md:11]: Hi @dakshdeepHERE I am having trouble understanding your request. When you say, add the tags to a tags.md file do you mean create a tags.md file?
Or are you saying to edit something within the linked-list.md file?

Comment (dakshdeepHERE) [reply content/c/concepts/linked-list/linked-list.md:11]: @Carloscode5 the `tags` should come from [here](https://github.com/Codecademy/docs/blob/main/documentation/tags.md) only i.e. `tags.md`. If the tags you used are not there then go to `tags.md` and add the tags that you've used here in the file. Sorry if I was not clear before @Carloscode5. Is there anything else I can help you with?

**Comment by dakshdeepHERE** [comment]
@Carloscode5 you have requested my review. But you haven't yet committed the changes I've requested before. Can you please work on them?

**Comment by Carloscode5** [comment]
Hi Daksha,

I believe I have committed the changes to all the requests.

Was there something that I was missing on my side? If so, can you help
guide me through the process?
When you say committed the changes you mean?

Best,

Carlos Valenzuela

**Comment by dakshdeepHERE** [comment]
@Carloscode5 Are you sure you pushed the changes from your side? Check the file changes once. None of my comments has been implemented.

**Comment by Carloscode5** [comment]
Hi Dak,I’ll try once again, just to confirm, you add all files, git commit and then git push from remote branch to main branch, which would end up in the repository through GitHub?Best,Carlos ValenzuelaOn Nov 18, 2023, at 10:20 AM, Daksha Deep ***@***.***> wrote:﻿
@Carloscode5 Are you sure you pushed the changes from your side? Check the file changes once. None of my comments has been implemented.

**Comment by dakshdeepHERE** [comment]
@Carloscode5 make sure you do the changes and push the changes in your `docs-issue-linked-list` branch. I still can't see your changes. See:
image

**Comment by Carloscode5** [comment]
Ah okay, I think I may have done it. I made the commits, but don’t think I Git Push’d them.They should be good to go now. Let me know if I can make any other adjustments.Best,Carlos ValenzuelaOn Nov 21, 2023, at 2:30 PM, Daksha Deep ***@***.***> wrote:﻿
@Carloscode5 make sure you do the changes and push the changes in your docs-issue-linked-list branch. I still can't see your changes. See:

**Comment by Carloscode5** [comment]
Ah, okay. Thank you for your patience. I’m going to look at them in a moment and go through them step by step.Best,Carlos ValenzuelaOn Nov 27, 2023, at 11:22 AM, Daksha Deep ***@***.***> wrote:﻿
No @Carloscode5, my requested changes are still there. Once open your PR, go to the files changed section and check the comments yourself they are still there. None of them have been implemented.
image.png (view on web)

**Comment by Carloscode5** [comment]
Hi ther, okay let me know if you see it now, I went ahead and created the PR. Let me know if that looks correct Dak.
 
Screen Shot 2023-11-27 at 2 39 42 PM

**Comment by SSwiniarski** [comment]
Closing this PR as the author submitted a duplicate PR.

**Pull request closed without merging**